com.qq.QQ.yaml

```diff
@@ -0,0 +1,80 @@
+app-id: com.qq.QQ
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+command: qq
+separate-locales: false
+
+finish-args:
+  - '--share=ipc'
+  - '--share=network'
+  - '--socket=x11'
+  - '--socket=pulseaudio'
+  - '--device=all'
+  - '--filesystem=xdg-documents:ro'
+  - '--filesystem=xdg-pictures:ro'
+  - '--filesystem=xdg-videos:ro'
+  - '--filesystem=xdg-download'
+  - '--talk-name=org.freedesktop.Notifications'
+  - '--talk-name=org.kde.StatusNotifierWatcher'
+  - '--system-talk-name=org.freedesktop.login1'
+  - '--filesystem=xdg-run/pipewire-0'
+
+cleanup:
+  - /include
+  - /lib/pkgconfig
+  - /share/gtk-doc
+  - '*.la'
+
+modules:
+  - name: qq
+    buildsystem: simple
+    build-commands:
+      - install apply_extra ${FLATPAK_DEST}/bin
+      - install -Dm755 qq.sh ${FLATPAK_DEST}/bin/qq
+      - install -Dm644 com.qq.QQ.metainfo.xml $FLATPAK_DEST/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - install -Dm755 /usr/bin/ar -t ${FLATPAK_DEST}/bin
+      - install -Dm755 /usr/bin/desktop-file-edit -t ${FLATPAK_DEST}/bin
+      - install -Dm755 /usr/lib/$(gcc -print-multiarch)/libbfd-*.so -t ${FLATPAK_DEST}/lib
+    sources:
+      - type: script
+        commands:
+          - export TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"
+          - exec zypak-wrapper /app/extra/QQ/qq "$@"
+        dest-filename: qq.sh
+
+      - type: script
+        commands:
+          - ar x qq.deb
+          - tar -xf data.tar.xz
+          - mv opt/QQ .
+          - mv usr export
+          - rm -rf export/share/doc
+          - rm -rf export/share/icons
+          - install -Dm644 QQ/resources/app/512x512.png export/share/icons/hicolor/512x512/apps/qq.png
+          - FLATPAK_ID=com.qq.QQ
+          - desktop-file-edit --set-icon="$FLATPAK_ID" --set-key="Exec" --set-value="qq %U" export/share/applications/*.desktop
+          - rename "qq" "$FLATPAK_ID" export/share/icons/hicolor/*/apps/*.png export/share/applications/*.desktop
+          - rm -rf qq.deb opt usr control.tar.gz data.tar.xz debian-binary
+        dest-filename: apply_extra
+      - type: file
+        path: com.qq.QQ.metainfo.xml
+      - type: extra-data
+        filename: qq.deb
+        url: https://dldir1.qq.com/qqfile/qq/QQNT/4691a571/QQ-v2.0.1-429_x64.deb
+        sha256: e3aa15ff6ae089b655df3913c6020ca56726bd676995aaf29b313fbd0643ea42
+        size: 108719314
+        # Will enable when available
+        # x-checker-data:
+        #     type: html
+        #     url: https://im.qq.com/linuxqq/download.html
+        #     version-pattern: '(\S*)'
+        #     url-template: 'https://dldir1.qq.com/qqfile/qq/QQNT/4691a571/QQ-(\S*)_x64.deb'
+  - name: kerberos
```

Comment:
this module should be before the "main" one as it is a dependency.

Reply:
Yeah, although it probably doesn't matter much as it repackages the binary.